### ISSUE TITLE
Comparing FID score fairly?

### ISSUE THREAD

**Issue by curiousbyte19** [lfovia/QAGANS#1]
Hello, thank you for the code. Unfortunately, the performance seem  **very different** from the paper comparison, and **most of the improvements come from using different number of generated images for computing FID.** 

50K generated samples for FID means we can get a very low FID score, and when you compare to SN-GAN, the official code use only 5K generated sample. Running same chainer code for SN-GAN using 50K samples is simple, and doing this with 50K samples I get **6 to 7 points of FID improvement ** (you can run the chainer models again and check). Since the authors use the original chainer implementation, maybe you can rerun the results to compare them fairly and update the score for Table 1.

https://papers.nips.cc/paper/8560-quality-aware-generative-adversarial-networks.pdf

**Comment by parimala1175** [comment]
[Response_Issue1.pdf](https://github.com/lfovia/QAGANS/files/4328790/Response_Issue1.pdf)

**Comment by lfovia** [comment]
Please find the attached pdf. Thank you.
[Response_Issue1.pdf](https://github.com/lfovia/QAGANS/files/4328934/Response_Issue1.pdf)

**Comment by curiousbyte19** [comment]
Thank you for the reply. To address your points:

1. I don't follow the logic in point 1. What specific pre-trained model do you refer to? The only pre-trained model for FID is inception model pretrained with imagenet.

2. In SN-GAN, the authors used 10K real images, also 5K fake images. Do you use the entire training dataset to compute the real image mean and cov? because this will give you a very good score. In table 3, your STL-10 scores is too good to be true, and better than any reported score in literature, you can correct me if this is not true.

3. To put it easily, table 3 is comparing (10k real, 5k gen) from SN-GAN with (100K real, 5k gen) from QA-GAN and table 4 is comparing (10k real, 5k gen) from SN-GAN with (50k real, 5k gen) from QA-GAN. This comparison is inaccurate and biased towards your model. Since you run the author's code for SN-GAN, why not compare them with same no. samples? In your table 3 and 4, instead of doing this, you still quoted the same score from the paper, which has different method to evaluate. Table 4, CIFAR-10 does not have 48 x 48.

4. My explanation on top follow your Table 5. You see just by using 50K gen vs 5K gen img, you have **4 points extra improvement**. The same for real image sample size, which you should use 10K if you want to compare fairly, or else should run SN-GAN with same real sample size too.

5. Only inception score (table 1 and 2) is accurate, as the method to compute is the same using 5K images. In this case, the performance is not significantly different from SN-GAN. 

6. Table 5, better to report mean score not single score, which can be from some best random seed. Won't be fair if comparing best score from 1 seed to worst score from another seed.

My conclusion remain the same. Most of the improvement in the paper come from changing fid computing method using sample size with non-accurate comparison of results. I don't blame you for the comparison done but hope you can adjust your score to correctly reflect the real results. Otherwise, it is very difficult for people to follow up in research to compare with your work. It is impotant to compare fair.

**Comment by curiousbyte19** [comment]
Thank you for the reply. To address your points:

1. What specific pre-trained model do you refer to? The only pre-trained model for FID is inception model pretrained with imagenet.

2. In SN-GAN, the authors used 10K real images, also 5K fake images. Do you use the entire training dataset to compute the real image mean and cov? because this will give you a very good score. In table 3, your STL-10 scores is too good to be true, and better than any reported score in literature, you can correct me if this is not true.

3. Table 3 is comparing (10k real, 5k gen) from SN-GAN with (100K real, 5k gen) from QA-GAN and table 4 is comparing (10k real, 5k gen) from SN-GAN with (50k real, 5k gen) from QA-GAN. This comparison is inaccurate and biased towards your model. Since you run the author's code for SN-GAN, you should compare with same no. samples. In your table 3 and 4, instead of doing this, you still quoted the same score from the paper, which has different method to evaluate. Table 4, CIFAR-10 does not have 48 x 48.

4. My explanation on top follow your Table 5. Just by using 50K gen vs 5K gen img, you have 4 points extra improvement. The same for real image sample size, which you should use 10K if you want to compare fairly, or else should run SN-GAN with same real sample size too.

5. Only inception score (table 1 and 2) is accurate, as the method to compute is the same using 5K images. Thanks for the producing this.

6. Table 5, better to report mean score not single score, which can be from some best random seed. Won't be fair if comparing best score from 1 seed to worst score from another seed.

My conclusion remain the same. Most of the improvement in the paper come from changing fid computing method using sample size with non-accurate comparison of results. I don't blame you for the comparison done but hope you can adjust your score to correctly reflect the real results. Otherwise, it is very difficult for people to follow up in research to compare with your work. It is impotant to compare fair.

thanks for giving back response and answering the issue. excellent effort to practice good science.

**Comment by lfovia** [comment]
Thanks for your comments on our response.
Please find the attached pdf. 
[Response_Issue1.pdf](https://github.com/lfovia/QAGANS/files/4379095/Response_Issue1.pdf)

We would happy to address any further concerns. Thanks.

**Comment by lfovia** [comment]
We would happy to answer any further concerns. Thank you.
